packages/editor-ui/src/components/ResourceLocator/ResourceLocator.vue

```diff
@@ -162,119 +236,425 @@ export default mixins().extend({
-			this.resourceIssues = this.parameterIssues.concat(validationErrors);
+		openResource(url: string) {
+			window.open(url, '_blank');
+			this.trackEvent('User clicked resource locator link');
 		},
-		findModeByName (name: string): INodePropertyMode | null {
+		getPropertyArgument(
+			parameter: INodePropertyMode,
+			argumentName: string,
+		): string | number | boolean | undefined {
+			if (parameter.typeOptions === undefined) {
+				return undefined;
+			}
+
+			// @ts-ignore
+			if (parameter.typeOptions[argumentName] === undefined) {
+				return undefined;
+			}
+
+			// @ts-ignore
+			return parameter.typeOptions[argumentName];
+		},
+		openCredential(): void {
+			const node = this.$store.getters.activeNode as INodeUi | null;
+			if (!node || !node.credentials) {
+				return;
+			}
+			const credentialKey = Object.keys(node.credentials)[0];
+			if (!credentialKey) {
+				return;
+			}
+			const id = node.credentials[credentialKey].id;
+			this.$store.dispatch('ui/openExisitngCredential', { id });
+		},
+		findModeByName(name: string): INodePropertyMode | null {
 			if (this.parameter.modes) {
 				return this.parameter.modes.find((mode: INodePropertyMode) => mode.name === name) || null;
 			}
 			return null;
 		},
-		getModeLabel (name: string): string | null {
-			return getParameterModeLabel(name);
+		getModeLabel(name: string): string | null {
+			if (name === 'id' || name === 'url' || name === 'list') {
+				return this.$locale.baseText(`resourceLocator.mode.${name}`);
+			}
+
+			return null;
+		},
+		onInputChange(value: string): void {
+			const params: INodeParameterResourceLocator = { value, mode: this.selectedMode };
+			if (this.isListMode) {
+				const resource = this.currentQueryResults.find((resource) => resource.value === value);
+				if (resource && resource.name) {
+					params.cachedResultName = resource.name;
+				}
+
+				if (resource && resource.url) {
+					params.cachedResultUrl = resource.url;
+				}
+			}
+			this.$emit('input', params);
 		},
-		onInputChange (value: string): void {
-			this.$emit('valueChanged', { value, mode: this.selectedMode });
+		onModeSelected(value: string): void {
+			if (typeof this.value !== 'object') {
+				this.$emit('input', { value: this.value, mode: value });
+			} else if (value === 'list') {
+				this.$emit('input', { value: '', mode: 'list' });
+			} else if (value === 'url' && this.value && this.value.cachedResultUrl) {
+				this.$emit('input', { mode: value, value: this.value.cachedResultUrl });
+			} else {
+				this.$emit('input', { mode: value, value: (this.value? this.value.value : '') });
+			}
+
+			this.trackEvent('User changed resource locator mode', { mode: value });
 		},
-		onModeSelected (value: string): void {
-			this.validate();
-			this.$emit('modeChanged', { mode: value, value: this.value });
+		trackEvent(event: string, params?: {[key: string]: string}): void {
```

Comment:
Not is scope: We should make this reusable.

Reply:
not sure. this is the only place it's used

Reply:
I don't see it right now, but we are tracking a lot of stuff on a lot of places so it may make sense. One question here: Do we need to explicitly add `instanceId` and `userId` since they are automatically sent with every telemetry event?